### PR TITLE
Run vtests on PRs and commits to master branch and compare to parent …

### PR DESCRIPTION
**Pull request by AntonioBL**
…commit vtest results; output a comment and upload a report when differences are found

By running vtests on each commit and PR we can early find when one if this tests is broken, or fixed. This Github action compares the vtests from the new commit with the results from the base (parent) commit, both compiled on the same machine.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [N/A] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [N/A] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by AntonioBL** [comment]
Here are examples of:
- a PR changing vtests: https://github.com/AntonioBL/MuseScore/actions/runs/43446918
- a push commit changin vtests: https://github.com/AntonioBL/MuseScore/actions/runs/43446862
- a PR which does not change vtests: https://github.com/AntonioBL/MuseScore/actions/runs/43459099

If automatically adding comments to PRs or commits becomes annoying, we could comment out that part of the code and simply issue a warning ( https://help.github.com/en/actions/reference/development-tools-for-github-actions#set-a-warning-message-warning )

**Comment by AntonioBL** [comment]
By the way, vtests at the moment need some cleaning.
Maybe continuously checking them will help in maintaining them and having a real benefit from having them.

**Comment by dmitrio95** [comment]
Actually we have [this page](http://vtest.musescore.org/compare.html) which allows comparing vtest results between different MuseScore revisions (by branch name or commit hash), so they definitely give a certain benefit now. But that doesn't work with pull requests as Travis doesn't allow uploading artifacts from pull request jobs, and it still requires manual checks. So having a way to detect differences automatically would certainly be helpful.

File: vtest/gen_compare [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# Close html file tags
+echo "  </body>" >> $F
+echo "</html>" >> $F
+
+# Remove "compare" folder if no differences were detected
+cd ..
+if [ "$FLAG" = false ]; then
+      rm -rf ./compare
+fi
+echo "::set-env name=FLAG::${FLAG}"
```
Comment (dmitrio95) [inline vtest/gen_compare]: Could this `FLAG` be renamed to something more meaningful, like `VTEST_DIFF_FOUND` or something like that? It looks like it is possible to give it any name.

Comment (AntonioBL) [reply vtest/gen_compare]: Ok, done. :+1:

**Comment by AntonioBL** [comment]
I think I found a misprint. I will update the PR in a couple of minutes.

**Comment by AntonioBL** [comment]
I had forgotten a space between "[" and "-z". Now it should be ok.

**Pull request merged**